### DISCUSSION TITLE
Applying Istio Telemetry config kills active connections across the kubernetes cluster

### DISCUSSION THREAD

**Discussion by dylanmcgrath** [istio/istio#52267] [Q&A] [answered]
I have a strange issue where when I apply Istio telemetry config, all pods in the cluster have all their active connections killed. This results in lots of applications losing their connections to RDS MySQL databases. This affects multiple environments.
Using Istio `1.21.2`

**Telemetry config**
```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  labels:
    argocd.argoproj.io/instance: istio-telemetry
  name: istio-access-logs
  namespace: istio-system
spec:
  accessLogging:
    - filter:
        expression: response.code >= 500
      providers:
        - name: custom-log
```

The applications connect via an ExternalName service that points at a R53 record targeting the relevant RDS database endpoint. There are 200+ ExternalNames if that matters.

My understanding of Istio Telemetry is that this should not happen, the envoy can accept config changes without reseting active connections.

In trying to get to the bottom of this I have really struggled to find a log that will explain why this is happening.

- Checked the `istio-proxy` sidecar containers before/after deploying and they do not restart
- Checking logs with `kubectl logs -n <ns> <podname> -c istio-proxy` on an affected pod shows no new messages around the time of the apply.
- Checking the `istiod` logs shows activity when I apply the config, starting with a `debounce` and followed by a lot of messages for all the pods in the cluster, but there no obvious indication that it will force a connection reset.
```
2024-07-23T13:58:16.309317Z info ads Push debounce stable[1733] 1 for config Telemetry/staging/istio-access-logs: 100.230966ms since last change, 100.230902ms since last push, full=true
2024-07-23T13:58:16.309464Z info ads XDS: Pushing Services:669 ConnectedEndpoints:172 Version:2024-07-23T13:58:16Z/277
2024-07-23T13:58:16.312341Z info ads LDS: PUSH for node:istio-ingressgateway-85b498756-6rnqg.istio-system resources:1 size:2.1kB
2024-07-23T13:58:16.341445Z info ads LDS: PUSH for node:app-a-549b444cdf-5v9t4.staging resources:239 size:538.4kB
2024-07-23T13:58:16.346882Z info ads LDS: PUSH for node:app-b-59745b7bcc-6tpwh.staging resources:171 size:714.0kB
2024-07-23T13:58:16.366661Z info ads LDS: PUSH for node:app-c-cc9f7bc69-tzv7c.staging resources:171 size:714.6kB
<snip>
```

If I make MySQL connections bypass the sidecar with the annotation `traffic.sidecar.istio.io/excludeOutboundPorts: "3306"` then as expected there are no connection drops, but I'd prefer to keep traffic within the mesh.

I would appreciate any explanation of why this is happening and/or how to troubleshoot the issue further.

**Comment by zirain** [comment, marked as answer]
Appling Telemetry config will cause the rebuild the listener, that worked as expected.

**Reply by dylanmcgrath** [reply]
Just to clarify, does a rebuild of the listener also mean outbound connections are affected?

**Reply by zirain** [reply]
Usually, yes

**Reply by dylanmcgrath** [reply]
Okay. Thank you for the response.
